tests/helpers_tests/test_condition.py

```diff
@@ -108,6 +126,273 @@ def test_evaluate__returns_null__when_condition_has_an_invalid_type_property(sel
 
         self.assertIsNone(evaluator.evaluate(0))
 
+    def test_evaluate__returns_true__when_user_version_2_matches_target_version_2(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_condition_list, {'Android': '2'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_2_2_matches_target_version_2(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_condition_list, {'Android': '2.2'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_2_0_matches_target_version_2_0(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_0_condition_list, {'Android': '2.0'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_2_0_0_matches_target_version_2_0_0(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_0_0_condition_list, {'Android': '2.0.0'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_false__when_user_version_2_0_does_not_match_target_version_2_0(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_0_0_condition_list, {'Android': '2.0'}, self.mock_client_logger
```

Comment:
How is this false?

Reply:
Same reason as  above. The target version has been defined to higher precision 2.0.0.  And the user version is only 2.0